### PR TITLE
Implements Scrollview, Volume, Effect Panel and Effect Dialog

### PR DESCRIPTION
**Pull request by Maxarreaga**

### CODE REVIEW

File: QT/sounddisplay.cpp [hunk 1]
```diff
@@ -78,4 +102,32 @@ void SoundDisplay::drawWaveFromFile(QString file)
...
+    }
+}
+
+void SoundDisplay::removeInputButtonPushed()
+{
+    this->setVisible(false);
+    // emit signal to inputScrollView to remove
+}
+
+void SoundDisplay::volumeChanged(int changedVolume)
```
Comment (luckuisha) [inline QT/sounddisplay.cpp:125]: change int to float

**Review by luckuisha — DISMISSED** [review]
fix the play button before merge. else looks good

File: QT/frequencydomaindisplay.cpp [hunk 2]
```diff
@@ -23,10 +23,22 @@ void FrequencyDomainDisplay::setCoefficients(QVector<double> coeffs)
     plot();
 }
 
+void FrequencyDomainDisplay::setSamples(QVector<double> samples)
+{
+
```
Comment (dudublad) [inline QT/frequencydomaindisplay.cpp]: Why is this here?

File: QT/recordedsounddisplay.cpp [hunk 3]
```diff
@@ -2,15 +2,24 @@
 
 // Uses parent constructor
 
-RecordedSoundDisplay::RecordedSoundDisplay(QWidget *parent) : SoundDisplay(parent)
+RecordedSoundDisplay::RecordedSoundDisplay(QWidget *parent, int id) : SoundDisplay(parent)
 {
     // init
+    inputId = id;
+    idLabel = new QLabel(this);
```
Comment (dudublad) [inline QT/recordedsounddisplay.cpp]: I don't think we should keep this

File: QT/sounddisplay.cpp [hunk 4]
```diff
@@ -7,31 +7,54 @@ SoundDisplay::SoundDisplay(QWidget *parent)
...
+    toggleEffectPanelButton = new QPushButton("Toggle Effects Panel",this);
     timeDomain = new TimeDomain(this);
     frequencyDisplay = new FrequencyDomainDisplay(this);
+    effectPanel = new EffectPanel(this);
+    idLabel = new QLabel(this);
+    volumeSlider = new QSlider(Qt::Horizontal,this);
+    volumeLabel = new QLabel(this);
+
+    // Volume Slider config
+    volumeSlider->setMaximum(100);
```
Comment (dudublad) [inline QT/sounddisplay.cpp:21]: volume slider is not working for recorded input

File: QT/sounddisplay.cpp [hunk 5]
```diff
@@ -40,9 +63,10 @@ SoundDisplay::~SoundDisplay()
     delete playButton;
     delete timeDomain;
     delete frequencyDisplay;
+    delete effectPanel;
     delete mainLayout;
-    delete domainLayout;
```
Comment (dudublad) [inline QT/sounddisplay.cpp:44]: Where does this get done? It needs to be deleted especially cus now we are adding and removing instances of SoundDisplay

File: QT/sounddisplay.cpp [hunk 6]
```diff
@@ -78,4 +102,32 @@ void SoundDisplay::drawWaveFromFile(QString file)
...
+    }
+    else
+    {
+        effectPanel->setVisible(false);
+    }
+}
+
+void SoundDisplay::removeInputButtonPushed()
+{
+    this->setVisible(false);
```
Comment (dudublad) [inline QT/sounddisplay.cpp:121]: When does the actual instance of the class get deleted?

File: QT/inputscrollview.cpp [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+    // This loop deletes inputs that are no longer valid
+    for(int i = 0;i < inputSubScrollLayout->count();i++)
+    {
+        //TODO: change this check to check for visibility
+        // as that is what we use
+        auto *widget = inputSubScrollLayout->itemAt(i)->widget();
+        auto widgetExists = inputs.contains(widget);
+        if(widgetExists == false)
+        {
+            inputSubScrollLayout->removeWidget(widget);
```
Comment (dudublad) [inline QT/inputscrollview.cpp:85]: delete WaveDisplay?
Update the vector?

File: QT/wavedisplay.cpp [hunk 8]
```diff
@@ -1,50 +1,75 @@
...
+    buttonLayout->addWidget(playButton,0,0,Qt::AlignCenter);
+    buttonLayout->addWidget(stopButton,1,0,Qt::AlignCenter);
+    buttonLayout->addWidget(toggleEffectPanelButton,2,0,Qt::AlignCenter);
+    buttonLayout->addWidget(titleLabel,0,1,Qt::AlignCenter);
+    buttonLayout->addWidget(frequencySlider,1,1,Qt::AlignCenter);
+    buttonLayout->addWidget(frequencyLabel,2,1,Qt::AlignCenter);
+    buttonLayout->addWidget(removeInputButton,3,0,Qt::AlignCenter);
+    buttonLayout->addWidget(waveTypeSelector,3,1,Qt::AlignCenter);
+    buttonLayout->addWidget(volumeLabel,4,0,Qt::AlignCenter);
+    buttonLayout->addWidget(volumeSlider,4,1,Qt::AlignCenter);
```
Comment (dudublad) [inline QT/wavedisplay.cpp:73]: This class needs a destructor

File: QT/wavedisplay.cpp [hunk 9]
```diff
@@ -109,3 +134,12 @@ void WaveDisplay::onPlayButtonClicked()
 {
     playSound();
 }
+
+void WaveDisplay::onSpinBoxChanged(double value)
```
Comment (dudublad) [inline QT/wavedisplay.cpp:138]: I think we need to limit the rate at which things get updated to prevent the program from crashing. I'll look into it

**Review by dudublad — DISMISSED** [review]

File: QT/sounddisplay.cpp (hunk 4 above)
Comment (luckuisha) [reply QT/sounddisplay.cpp:21]: figure this out later

**Pull request merged**